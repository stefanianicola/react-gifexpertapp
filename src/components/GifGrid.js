import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);


    return (
        <>
            <h3>{category}</h3>

            {
                loading ? (<p>Cargando...</p>) : (
                    <div className='card-grid'>

                        <ol>
                            {
                                images.map(img => {
                                    return <GifGridItem
                                        key={img.id}
                                        {...img} />
                                })
                            }
                        </ol>
                    </div>
                )
            }

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
