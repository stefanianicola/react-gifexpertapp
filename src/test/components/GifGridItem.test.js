import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en GifExpertApp', () => {
    const title = 'Titulo';
    const url = 'https://algo'

    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    it('debe mostrar bien el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('debe tener un p con su title', () => {

        const p = wrapper.find('p')

        expect(p.text().trim()).toBe(title)

    });

    it('debe tener una img con su url y su title iguales a los props', () => {

        const img = wrapper.find('img')

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

})