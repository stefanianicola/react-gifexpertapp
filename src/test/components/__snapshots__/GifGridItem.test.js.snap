// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp debe mostrar bien el componente 1`] = `
<div
  className="card"
>
  <img
    alt="Titulo"
    src="https://algo"
  />
  <p>
    Titulo
  </p>
</div>
`;
